tests/test_rio_info.py

```diff
@@ -8,6 +8,361 @@
+    runner = CliRunner()
+    inputfile = str(data.join('RGB.byte.tif'))
+    result = runner.invoke(
+        main_group, ['edit-info', inputfile, '--unset-nodata', '--nodata', '0'])
+    assert result.exit_code == 2
+
+
+def test_delete_crs_exclusive_opts(data):
+    """--unset-crs and --crs can't be used together"""
+    runner = CliRunner()
+    inputfile = str(data.join('RGB.byte.tif'))
+    result = runner.invoke(
+        main_group, ['edit-info', inputfile, '--unset-crs', '--crs', 'epsg:4326'])
+    assert result.exit_code == 2
+
+
+@pytest.mark.skip(
+    parse(rasterio.__gdal_version__) < parse('1.10'),
+    reason='GDAL version >= 1.10 required')
+def test_unset_crs(data):
+    runner = CliRunner()
+    inputfile = str(data.join('RGB.byte.tif'))
+    result = runner.invoke(main_group,
+                           ['edit-info', inputfile, '--unset-crs'])
+    assert result.exit_code == 0
+    with rasterio.open(inputfile) as src:
+        assert src.crs is None
+
+
+@pytest.mark.skip(
+    parse(rasterio.__gdal_version__) >= parse('1.10'),
+    reason='Test applies to GDAL version < 1.10')
+def test_unset_crs_gdal19(data):
+    """unsetting crs doesn't work for geotiff and gdal 1.9
+    and should emit an warning"""
+    runner = CliRunner()
+    inputfile = str(data.join('RGB.byte.tif'))
+    with rasterio.open(inputfile) as src:
+        orig_crs = src.crs
+    with pytest.warns(UserWarning):
+        result = runner.invoke(main_group,
+                               ['edit-info', inputfile, '--unset-crs'])
+    assert result.exit_code == 0
+    with rasterio.open(inputfile) as src:
+        assert src.crs == orig_crs  # nochange
+
+
+def test_edit_nodata_err(data):
+    runner = CliRunner()
+    inputfile = str(data.join('RGB.byte.tif'))
+    result = runner.invoke(main_group,
+                           ['edit-info', inputfile, '--nodata', '-1'])
+    assert result.exit_code == 2
+
+
+@pytest.mark.xfail(
+    Version(rasterio.__gdal_version__) < Version('2.1'),
+    reason='GDAL version >= 2.1 required')
+def test_delete_nodata(data):
+    """Delete a dataset's nodata value"""
+    runner = CliRunner()
+    inputfile = str(data.join('RGB.byte.tif'))
+    result = runner.invoke(
+        main_group, ['edit-info', inputfile, '--unset-nodata'])
+    assert result.exit_code == 0
+
+
+def test_edit_nodata(data):
+    runner = CliRunner()
+    inputfile = str(data.join('RGB.byte.tif'))
+    result = runner.invoke(
+        main_group, ['edit-info', inputfile, '--nodata', '255'])
+    assert result.exit_code == 0
+    with rasterio.open(inputfile) as src:
+        assert src.nodata == 255.0
+
+
+def test_edit_nodata_nan(data):
+    runner = CliRunner()
+    inputfile = str(data.join('float_nan.tif'))
+    result = runner.invoke(
+        main_group, ['edit-info', inputfile, '--nodata', 'NaN'])
+    assert result.exit_code == 0
+    with rasterio.open(inputfile) as src:
+        assert src.nodata != src.nodata
```

Comment:
How does this even work??  As written, this looks like a logic error (how can something not equal itself).  Is there a way to rewrite this to instead make sure it has the desired value set?

Reply:
This only shows up as new due to the test shuffle but its a byproduct of how `numpy.nan` operates.  `np.nan == np.nan == False` and `(np.nan != np.nan) == True`, but I replaced it with a `np.isnan(src.nodata)`, which makes _way_ more sense.